component/service/src/main/java/org/exoplatform/social/rest/impl/favorite/FavoriteRest.java

```diff
@@ -106,6 +112,45 @@ public Response createFavorite(
     }
   }
 
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(
+      value = "Retrieves all favorites object of the current user",
+      httpMethod = "GET",
+      response = Response.class,
+      produces = "application/json",
+      notes = "Return list of favorites"
+  )
+  @ApiResponses(
+      value = {
+          @ApiResponse(code = 200, message = "Request fulfilled"),
+          @ApiResponse(code = 500, message = "Internal server error")
+      }
+  )
+  public Response getFavoritesList(
+                                      @ApiParam(
+                                          value = "Query Offset",
+                                          required = true
+                                      )
+                                      @QueryParam("offset")
+                                      int offset,
+                                      @ApiParam(
+                                          value = "Query results limit",
+                                          required = true
+                                      )
+                                      @QueryParam("limit")
+                                      int limit ) {
+    long userIdentityId = RestUtils.getCurrentUserIdentityId();
```

Comment:
Could we add condition for offset and limit, offset must be 0 or positive and limit must be positive

Reply:
we add consition in front side https://github.com/Meeds-io/social/blob/Task-53938/webapp/portlet/src/main/webapp/vue-apps/common/js/FavoriteService.js#L2
and also in the DAO https://github.com/Meeds-io/social/blob/Task-53938/component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/MetadataItemDAO.java#L135